### DISCUSSION TITLE
Passing response_content_disposition to _url method

### DISCUSSION THREAD

**Discussion by majksner** [renchap/shrine-google_cloud_storage#51] [Q&A] [answered]
Hi there,

I was wondering if it is possible to pass `response_content_disposition` as a response header with Google Cloud Storage, this works perfectly with S3 Storage.

For example, when using S3:

```ruby
uploaded_file.file_url(response_content_disposition: ContentDisposition.attachment('file.txt'))
```

The above example works fine with S3. However, it does not work with Google Cloud Storage.

I couldn't find anything documented about it, except to pass 'Content-Dispositon' as a header at the time of upload via url_options plugin, but I want to pass it at the time of download so the file that is downloaded gets a different name than the file that is actually stored on the Google Cloud Storage.

Thanks!

**Comment by renchap** [comment]
Hi @majksner 

Sorry for the delay, I missed the notification for your question!

As far as I know, header responses can not be modified at response-time on GCS. You need to proxy the response and add the header yourself. 

For Rails, the best way would probably be to use open a stream from your file on GCS and use [`send_data`](https://edgeapi.rubyonrails.org/classes/ActionController/DataStreaming.html) to stream it back to the client with the correct header.

**Comment by majksner** [comment, marked as answer]
Hi,

I was able to do this by passing

```ruby
{
    query: {
    'response-content-disposition':
      ActionDispatch::Http::ContentDisposition.format(disposition: 'attachment', filename: 'file.txt')
    }
}
```

to `file_url()` method

**Reply by renchap** [reply]
Oh, I did not know about this `query` parameter and that you can change the response headers with it on GCS, this might be useful, thanks!
